app/components/session/home/home.controller.js

```diff
@@ -1,19 +1,54 @@
-phinisiApp.controller('detailsController', function($scope, $http){
-	$scope.storeDetails = {
-		store_name: '',
-		store_logo: '',
-		phone_number: '',
-		address: '',
-		province: 'Select your province...',
-		city: 'Select your city...'
-	}
+phinisiApp.controller('detailsController', ['$scope', '$http', '$window', '$log', 
+	function($scope, $http, $window, $log){
+		$scope.storeDetails = {
+			store_name: '',
+			store_logo: '',
+			phone_number: '',
+			address: '',
+			province: '',
+			city: ''
+		}
+		$scope.provinces = [{id:1, name:'Jawa Barat'}, {id:2, name: 'Banten'}];
+		$scope.cities = [{provinceId: 1, cityId: 1, name:'Bandung'}, {provinceId: 1, cityId: 2, name:'Bogor'}, {provinceId: 2, cityId: 3, name:'Tangerang'},];
+		$scope.$log = $log;
 
-	$scope.provinces = [ 'Jawa Barat', 'Jakarta', 'Sumatera Utara' ];
-
-	$scope.city = [ 'tes' ]
-
-	$scope.submit = function () {
-		
-	}
-
-})
+		$scope.submit = function(){
+				//get store details from model
+				console.log($window.sessionStorage.token);
+				console.log($scope.storeDetails);
+				$http.post(
+					//url
+					phinisiEndpoint + '/merchant/info/create',
+					//data
+					{
+					  token: $window.sessionStorage.token, 
+					  merchant_details: {
+					  	merchant_name : $scope.storeDetails.store_name, 
+					  	merchant_logo_url : $scope.storeDetails.store_logo
+					  }, 
+					  merchant_address: {
+					      city_id : $scope.storeDetails.city, 
+					      address : $scope.storeDetails.address, 
+					      phone_number : $scope.storeDetails.phone_number
+					  }
```

Comment:
You can make it shorter by make object for merchant_details and merchant_address separated. For example :

$scope.storeDetails = {
    merchant_details : {
        store_name : ' ',
        store_logo : ' '
    },
   merchant_address : {
          city_id : ' ',
          address : ' ',
          phone_number : ' '
    }
}

in this way you can put your http_post data variable like this :

merchant_details : $scope.storeDetails.merchant_details, merchant_address : $scope.storeDetails.merchant_address

even simpler, you can just put it like this
 $scope.storeDetails

because my interceptor also automatically create json object based on that object. If the variable's name already has same format as contract from API then it will automatically create it.